ascent_macro/src/ascent_codegen.rs

```diff
@@ -212,11 +212,29 @@ pub(crate) fn compile_mir(mir: &AscentMir, is_ascent_run: bool) -> proc_macro2::
    let relation_initializations_for_default_impl = 
       if is_ascent_run {vec![]} else {relation_initializations};
    let summary = mir_summary(mir);
-   let (impl_generics, ty_generics, where_clause) = mir.declaration.generics.split_for_impl();
-   let vis = &mir.declaration.visibility;
-   let struct_name = &mir.declaration.ident;
-   let generics = &mir.declaration.generics;
-   let struct_attrs = &mir.declaration.attrs;
+   let ty_signature = &mir.signatures.declaration;
+   let impl_signature = &mir.signatures.implementation;
+   
+   let (ty_impl_generics, ty_ty_generics, ty_where_clause) = ty_signature.generics.split_for_impl();

```

Comment:
Could you add a function `split_for_impl()` to `Signatures` and move this logic there?
Also please turn panics into compile errors.

Reply:
Done.